### DISCUSSION TITLE
Is it possible to disconnect users?

### DISCUSSION THREAD

**Discussion by mway** [zellij-org/zellij#2577] [Q&A]
Context: I often use zellij over ssh, for example:

```
# ssh -t my-server "zellij a -c dev"
```

This works great, except for if I get disconnected from SSH (e.g. my wifi drops, or some other intermittent network issue). When I get disconnected, sometimes reconnecting will create a second copy of the session that I'm attaching to (e.g., it thinks that a second user is joining the session).

This new session doesn't appear in `zellij ls`, but e.g. will change the UI to indicate that another user is connected.

This in itself isn't a big deal, but I don't like that it
1. adds cursors for the now-defunct older session that take some time (unclear how long, but a while) to disappear; and
2. changes the UI in sometimes weird ways (e.g. colors become different, borders change, etc)

What I'd like to do is force disconnect/close all other copies of the same session, basically reducing the "current users" to the current user that I'm operating as.

Is this possible?

**Comment by jaeheonji** [comment]
> What I'd like to do is force disconnect/close all other copies of the same session, basically reducing the "current users" to the current user that I'm operating as.

Unfortunately, no action is currently defined to control users connected to the session. And as far as I know, zellij doesn't internally define user as a unique value. Since users are simply registered in the order of connection, it would be very difficult to forcibly disconnect a specific user.

**Reply by Lockszmith-GH** [reply]
what about just: `killall zellij` ?
or am I missing something?

Also, I found that naming sessions works best for reconnecting. So not relying on the random naming Zellij defaults to.

**Comment by mway** [comment]
> what about just: `killall zellij` ?

If my goal is to continue to use the session, just without multiple users (because it thinks my SSH reconnection is a "new user"), then this won't really help, unless I'm misunderstanding.

I'll just experiment with mirroring sessions and see if that helps.

**Reply by Lockszmith-GH** [reply]
right, (note, there is a thread-reply (click the 'Write a reply) so you can have 'cleaner conversations'.

If you login with the same user credentials, you should have access to all processes started by that user.
I can only think of one reason that wouldn't be the case - if you run zellij as sudo, then the user id might be different (0 instead of your user's id)

To be sure about what's going on, track the user id (run `id` from the command line). And see if that changes across sessions (it shouldn't)

**Reply by dechimp** [reply]
When I attach to a running zellij session from different terminal instances (as the same user), zellij is showing each connection as a different user.   I often would like to disconnect the other users because 1) they are me and 2) they are interfering with various things like window resizing, etc.  And killing the session makes you have to start over, which is the pits.

I noticed that the Plugin API has a [command](https://zellij.dev/documentation/plugin-api-commands.html?highlight=disconnect_#disconnect_other_clients): `disconnect_other_clients`
```
Requires the ChangeApplicationState permission

Disconnect all other clients attached to this session except the one making this call
```

So maybe this functionality can be plugin-ized if it hasn't already?

**Reply by dechimp** [reply]
Sorry, this has already been implemented with `Ctrl-x` in the session manager https://x.com/Zellij_dev/status/1785911597795512468

**Reply by mpr1255** [reply]
Have been hunting for this for *ages*, and finally found this solution.  However, ctrl+x doesn't do anything, either from ctrl+o OR ctrl+o then W. What am I missing? I also get the resize limitation issue and have come up with crazy workarounds

**Reply by mpr1255** [reply]
Update: It works in 0.40.1 (on my mac); but does *not* appear to work on 0.42.0 (ubuntu, remote). No clue what accounts for this, or if I've simply does something wrong, or it's some super obscure escape key issue, where my actual keypress is being intercepted etc.

**Reply by ethauvin** [reply]
> Update: It works in 0.40.1 (on my mac); but does _not_ appear to work on 0.42.0 (ubuntu, remote). No clue what accounts for this, or if I've simply does something wrong, or it's some super obscure escape key issue, where my actual keypress is being intercepted etc.

Having the same issue with Arch/Manjaro and Ubuntu. I suspect it is because `ctrl-x` is used by the clipboard manager. I couldn't easily figure out how to change the key binding.
